opengever/maintenance/scripts/repository_migration.py

```diff
@@ -219,6 +284,55 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.patch_refs(DisabledLDAP, '__exit__', __exit__)
 
 
+class PatchCatalogMetadataUpdate(MonkeyPatch):
+    """The ZCatalog will always update all available metadata if reindexing an
+    object with 'update_metadata' enabled. There is no way to update only specific
+    metadata values of an object. Updating metadata means, that all metadata
+    will be recalcuated for an objects. This process takes a lot of time and is mostly
+    unnecessary. In addition, it requires to have access to the blobs of documents
+    because some of the metadata properties are relying on blob information (filesize).
+
+    To speed up the migration and to be able to run the migration without the
+    existence of blobs, we'll patch the catalog to only update the relevant
+    metadata items.
+    """
+
+    def __call__(self):
+        from Products.ZCatalog.Catalog import Catalog
+        from Products.ZCatalog.Catalog import safe_callable
+        from Products.ZCatalog.Catalog import MV
+
+        def recordify(self, obj):
+            # --- Patch ---
+            assert set(self.names) == EXPECTED_CATALOG_METADATA_FIELDS, \
```

Comment:
I would do this only once, not for each recordify call.
Maybe before patching?

Reply:
Oh yes... sure! Thanks!